client/web: test failure + data race

When I fixed https://github.com/tailscale/tailscale/pull/9149, this test started failing.

cc @soniaappasamy @willnorris @bradfitz 
```
➜  tailscale.com git:(maisem/flake-timeout) go test ./client/web -race
2023/08/29 18:22:37 Cannot parse QNAP login URL <nil>
2023/08/29 18:22:37 Cannot parse QNAP login URL parse "http://192.168.0.%31/": invalid URL escape "%31"
PASS
==================
WARNING: DATA RACE
Read at 0x00c0000ddd83 by goroutine 16:
  testing.(*common).logDepth()
      testing/testing.go:1011 +0xbb
  testing.(*common).log()
      testing/testing.go:1004 +0x7d
  testing.(*common).Error()
      testing/testing.go:1061 +0x55
  tailscale.com/client/web.TestServeAPI.func1()
      tailscale.com/client/web/web_test.go:88 +0x1b3

Previous write at 0x00c0000ddd83 by goroutine 15:
  testing.tRunner.func1()
      testing/testing.go:1582 +0x81a
  runtime.deferreturn()
      runtime/panic.go:477 +0x30
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44

Goroutine 16 (running) created at:
  tailscale.com/client/web.TestServeAPI()
      tailscale.com/client/web/web_test.go:82 +0x270
  testing.tRunner()
      testing/testing.go:1595 +0x238
  testing.(*T).Run.func1()
      testing/testing.go:1648 +0x44

Goroutine 15 (finished) created at:
  testing.(*T).Run()
      testing/testing.go:1648 +0x82a
  testing.runTests.func1()
      testing/testing.go:2054 +0x84
  testing.tRunner()
      testing/testing.go:1595 +0x238
  testing.runTests()
      testing/testing.go:2052 +0x896
  testing.(*M).Run()
      testing/testing.go:1925 +0xb57
  main.main()
      _testmain.go:49 +0x2bd
==================
panic: Log in goroutine after TestServeAPI has completed: use of closed network connection


goroutine 15 [running]:
testing.(*common).logDepth(0xc0000ddd40, {0xc000122030, 0x21}, 0x3)
        testing/testing.go:1022 +0x674
testing.(*common).log(...)
        testing/testing.go:1004
testing.(*common).Error(0xc0000ddd40, {0xc000089fc0, 0x1, 0x1})
        testing/testing.go:1061 +0x7e
tailscale.com/client/web.TestServeAPI.func1()
        tailscale.com/client/web/web_test.go:88 +0x1b4
created by tailscale.com/client/web.TestServeAPI in goroutine 14
        tailscale.com/client/web/web_test.go:82 +0x271
FAIL    tailscale.com/client/web        0.011s
FAIL
```

--- a/client/web/web_test.go
+++ b/client/web/web_test.go
@@ -77,17 +77,13 @@ func TestQnapAuthnURL(t *testing.T) {
 func TestServeAPI(t *testing.T) {
 	lal := memnet.Listen("local-tailscaled.sock:80")
 	defer lal.Close()
-
 	// Serve dummy localapi. Just returns "success".
-	go func() {
-		localapi := &http.Server{Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprintf(w, "success")
-		})}
-		defer localapi.Close()
-		if err := localapi.Serve(lal); err != nil {
-			t.Error(err)
-		}
-	}()
+	localapi := &http.Server{Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "success")
+	})}
+	defer localapi.Close()
+
+	go localapi.Serve(lal)
 	s := &Server{lc: &tailscale.LocalClient{Dial: lal.Dial}}
 
 	tests := []struct {

